src/KDSoapClient/KDSoapClientInterface.cpp

```diff
@@ -135,23 +137,40 @@ QNetworkRequest KDSoapClientInterfacePrivate::prepareRequest(const QString &meth
     return request;
 }
 
-QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const KDSoapHeaders &headers)
+QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const QString &soapAction, const KDSoapHeaders &headers)
 {
     KDSoapMessageWriter msgWriter;
     msgWriter.setMessageNamespace(m_messageNamespace);
     msgWriter.setVersion(m_version);
     const QByteArray data = msgWriter.messageToXml(message, (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(), headers,
                                                    m_persistentHeaders, m_authentication);
     QBuffer *buffer = new QBuffer;
-    buffer->setData(data);
+    if (m_sendSOAP_ActionInWS_AddressingHeader) {
+        KDSoapMessage messageCopy = message;
+        KDSoapMessageAddressingProperties prop = message.messageAddressingProperties();
+        if (!prop.action().isEmpty())
+            qWarning("Overwriting the action addressing parameter (%s) with the SOAP action (%s)",
+                     prop.action().toLocal8Bit().constData(), soapAction.toLocal8Bit().constData());
+        prop.setAction(soapAction);
+        messageCopy.setMessageAddressingProperties(prop);
+        buffer->setData(msgWriter.messageToXml(messageCopy,
```

Comment:
This 4-lines statement could be a local lambda, to avoid the code duplication. It would ease future maintenance.

Reply:
Done